### DISCUSSION TITLE
certificados

### DISCUSSION THREAD

**Discussion by GabrielIFPB** [qualitaocupacional/libesocial#11] [General]
Oi, boa tarde, estou com dúvidas sobre os certificados, pode me ajudar?

no seu arquivos .pem tem 4 certificados, quais são?

no site do esocial mostra 3 que deve ser baixados e deve ser colocado na ordem em que mostra a documentação

A instalação das cadeias deve seguir a ordem que consta nos endereços:

https://certificados.serpro.gov.br/serproacf/certificate-chain
https://repositorio.serpro.gov.br/docs/icpbrasilv2.crt
https://repositorio.serpro.gov.br/docs/acrfbv3.crt
https://repositorio.serpro.gov.br/docs/acserprorfbv4.crt

**Comment by andidevel** [comment]
> Oi, boa tarde, estou com dúvidas sobre os certificados, pode me ajudar?
>

Olá! Posso tentar ajudar sim, :).
 
> no seu arquivos .pem tem 4 certificados, quais são?
>

Se me lembro bem, 3 são esses aqui que você mencionou:

https://repositorio.serpro.gov.br/docs/icpbrasilv2.crt 
https://repositorio.serpro.gov.br/docs/acrfbv3.crt 
https://repositorio.serpro.gov.br/docs/acserprorfbv4.crt

Ou a versão válida desses certificados, pois lembro que o manual do eSocial estava errado com relação a pelo menos um certificado (veja o [README](https://github.com/qualitaocupacional/libesocial/tree/develop#certificados-do-icp-brasil-no-lado-cliente) do projeto. O quarto certificado, se não me engano foi um teste de conexão que estávamos fazendo e queríamos ver quais certificados iriam funcionar. Mas acho que ele não é usado (talvez não agora, quem sabe no futuro).

> no site do esocial mostra 3 que deve ser baixados e deve ser colocado na ordem em que mostra a documentação 
> A instalação das cadeias deve seguir a ordem que consta nos endereços:
> 

Não estou muito certo se a ordem importa, talvez se você for instalar os certificados um por um no sistema operacional. De qualquer forma, com a LIBeSocial [você não precisa instalar nenhum certificado](https://github.com/qualitaocupacional/libesocial/tree/develop#certificados-do-icp-brasil-no-lado-cliente) no sistema operacional que tem o aplicativo que vai se comunicar com o eSocial. Esse arquivo `.pem` que vem com a lib é "linkado" em cada conexão com os servidores do eSocial e os certificados ficam disponíveis para validar a conexão do servidor com o cliente.

Espero ter ajudado.
